packages/dds/tree/src/feature-libraries/treeTextCursor.ts

```diff
@@ -126,7 +126,7 @@ export class TextCursor implements ITreeCursor {
         this.siblings = this.parentStack.length === 0 ?
             this.root :
             (this.parentStack[this.parentStack.length - 1].fields ?? {}
-                )[this.keyStack[this.keyStack.length - 1] as string];
+                )[this.keyStack[this.keyStack.length - 1] as string] || {};
```

Comment:
I'm not sure what you are trying to do here, but use of "||" on something other than a boolean is concerning.

If you want to handle an undefined value here, use `??` instead.

Also the type of this expression is array of JsonableTree, so replacing the undefined case (which the type system doesn't acknowledge could happen, but could due to out of bounds access if keyStack is empty) with an empty object seems wrong. That both has the wrong type, and should never happen if the cursor is implemented correctly (I think).

It really annoys me that sticking asserts in there hurt perf for production usecases: I'd like a debug only assert library to make code like this more robust.



Anyway, if this is a bug in text cursor (and it looks like it might be), we should include a regression test making it clear it is a bug and that this fixes it. There is a test suite for this cursor.

Reply:
Using `||` would achieve the same result in this case.  However, you are right using `??` would be safer.